STYLE_GUIDE.md

```diff
@@ -0,0 +1,461 @@
+# Pony Standard Library Style Guide
+
+This is a document describing the preferred practices for writing code for the Pony standard library. Other projects are free to use the same style, but you are free to use any style in other projects. The following conventions are meant to ensure consistency and readability of Pony code. Any of the following rules may be broken in situations where the recommended practice would diminish readability.
+
+
+## Formatting
+
+### Line Length
+
+The maximum line length is 80 columns.
+
+### Trailing Whitespace
+
+No trailing whitespace at the end of lines.
+
+### Indentation
+
+Use 2 spaces for indentation, No hard tabs.
+
+```pony
+actor Main
+  new create(env: Env) =>
+    env.out.print("Hello, world!")
+```
+
+### Blank Lines
+
+- One blank line is placed between top-level declarations such as classes, primitives, actors, etc.
+- There is no blank lines between the class declaration and its first member or documentation.
+- Fields have no blank lines between them.
+- Functions are separated from other members by one line.
+
+  ```pony
+  // OK
+  class Foo
+  let bar: Bar
+  let baz: Baz
+
+  fun foo() =>
+  // ...
+
+  fun bar() =>
+  // ...
+
+  // OK
+  class Foo
+  """
+  Some documentation
+  """
+  let bar: Bar
+
+  fun foo() =>
+  // ...
+
+  // OK
+  class Foo
+  fun foo() =>
+  // ...
+
+  // Not OK
+  class Foo
+
+  let bar: Bar
+  let baz: Baz
+  fun foo() =>
+  // ...
+  ```
+
+- Blank lines between __one-line__ functions or primitives may be omitted if they are related.
+
+  ```pony
+  primitive Red fun apply(): U32 => 0xFF0000FF
+  primitive Green fun apply(): U32 => 0x00FF00FF
+  primitive Blue fun apply(): U32 => 0x0000FFFF
+
+  type Color is (Red | Green | Blue)
+  ```
+
+### Whitespace
+
+- Use spaces around binary operators and after commas, colons, and semicolons. Do not put spaces around brackets or parentheses.
+
```

Comment:
Theodus can you update to include that?

Please no force push that will kill off comments early though.